File: lc/c/030.c
int maxProfit(int* prices, int pricesSize){
    int max_profit = 0, min_so_far = INT_MAX;
    for (int i = 0; i < pricesSize; ++i) {
        int day_profit = prices[i] - min_so_far;
        max_profit = day_profit > max_profit ? day_profit : max_profit;
        min_so_far = prices[i] < min_so_far ? prices[i] : min_so_far;
    }
    return max_profit;
}

File: lc/py/030.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit, min_so_far = 0, float('inf')
        for p in prices:
            day_profit = p - min_so_far
            max_profit = max(max_profit, day_profit)
            min_so_far = min(min_so_far, p)
        return max_profit
